include/net/net_if.h

```diff
@@ -1950,7 +1950,7 @@ bool net_if_need_calc_tx_checksum(struct net_if *iface);
  *
  * @return Pointer to interface or NULL if not found.
  */
-struct net_if *net_if_get_by_index(int index);
+__syscall struct net_if *net_if_get_by_index(int index);
```

Comment:
Might want to add to the documentation here that this is a syscall to provide access to the object for purposes of assigning permissions.